libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -384,7 +400,8 @@ static void eventCallback( MQTTContext_t * pContext,
     assert( pContext != NULL );
     assert( pPacketInfo != NULL );
 
-    if( pPacketInfo->type == disconnectOnPacketType )
+    if( ( pPacketInfo->type == disconnectOnPacketType ) ||
```

Comment:
Nit: You could just have `if( ( pPacketInfo->type & 0xF0U ) == ( disconnectOnPacketType & 0xF0U ) )` if you want a shorter condition